lib/_stream_readable.js

```diff
@@ -72,7 +72,7 @@ function ReadableState(options, stream) {
   this.highWaterMark = (hwm || hwm === 0) ? hwm : defaultHwm;
 
   // cast to ints.
-  this.highWaterMark = ~~this.highWaterMark;
+  this.highWaterMark = Math.trunc(this.highWaterMark);
```

Comment:
Actually, I think we may want to use `Math.floor()` instead, it is significantly faster (at least on V8 5.8).